### PR TITLE
Fix bugs in debugger

### PR DESCRIPTION
**Pull request by clover2123**
* fix debugger to add breakpoint at the first line
* fix debugger not to insert any breakpoint when compiling dynamic code
* add some assertion code to detect undesirable cases

Signed-off-by: HyukWoo Park <hyukwoo.park@samsung.com>

### CODE REVIEW

**Comment by clover2123** [comment]
This patch fixed two errors.
1) breakpoint insertion is not allowed when its line number is `1`.
2) ByteCodeGenerator should not insert any breakpoint for dynamic code (dynamically created source code), but during the backtrace operation, ByteCodeGenerator is invoked again to get the bytecode location information and it inserts breakpoints at this time, so it may trigger another error.

File: src/parser/ast/ReturnStatementNode.h [hunk 1]
```diff
@@ -36,22 +36,14 @@ class ReturnStatementNode : public StatementNode {
     virtual void generateStatementByteCode(ByteCodeBlock* codeBlock, ByteCodeGenerateContext* context) override
     {
 #ifdef ESCARGOT_DEBUGGER
-        if (context->m_breakpointContext->m_breakpointLocations->size() == 0 && context->m_breakpointContext->m_parsingEnabled) {
-            ASSERT(context->m_breakpointContext->m_lastBreakpointLineOffset == 0);
-            ASSERT(context->m_breakpointContext->m_breakpointLocations->size() == 0);
-
+        {
```
Comment (zherczeg) [inline src/parser/ast/ReturnStatementNode.h:39]: Do we need this `{}` block?

**Review by zherczeg — COMMENTED** [review]
LGTM (informally)

File: src/parser/ast/ReturnStatementNode.h (hunk 1 above)
Comment (clover2123) [reply src/parser/ast/ReturnStatementNode.h:39]: No, it's just a matter of taste.
I want to seperate debugger-relate part clearly from other code.

**Review by ksh8281 — APPROVED** [review]
Nice!

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**